src/admin/runners/fixDuplicatePractitionerRunner.js

```diff
@@ -0,0 +1,505 @@
+        mongoCollectionManager,
+        collections,
+        batchSize,
+        deleteData,
+        adminLogger,
+        mongoDatabaseManager,
+        startFromCollection,
+        limit,
+        skip,
+        startFromId,
+        useTransaction,
+        properties,
+        afterLastUpdatedDate,
+        beforeLastUpdatedDate
+    }) {
+        super({
+            mongoCollectionManager,
+            batchSize,
+            adminLogger,
+            mongoDatabaseManager
+        });
+        /**
+         * @type {string[]}
+         */
+        this.collections = collections;
+
+        /**
+         * @type {string|undefined}
+         */
+        this.startFromCollection = startFromCollection;
+
+        /**
+         * @type {number|undefined}
+         */
+        this.limit = limit;
+
+        /**
+         * @type {string|undefined}
+         */
+        this.skip = skip;
+
+        /**
+         * @type {string|undefined}
+         */
+        this.startFromId = startFromId;
+
+        /**
+         * @type {boolean|undefined}
+         */
+        this.useTransaction = useTransaction;
+
+        /**
+         * @type {string[]|undefined}
+         */
+        this.properties = properties;
+
+        /**
+         * @type {string|undefined}
+         */
+        this.afterLastUpdatedDate = afterLastUpdatedDate;
+
+        /**
+         * @type {string|undefined}
+         */
+        this.beforeLastUpdatedDate = beforeLastUpdatedDate;
+
+        /**
+         * @type {string|undefined}
+         */
+        this.deleteData = deleteData;
+
+        /**
+         * @type {Array}
+         */
+        this.dupUuids = [];
+
+        /**
+         * stores resources and fields to be updated
+         * @type {Map<string, string[]>}
+         */
+        this.fieldsToUpdate = new Map([
+            ['Appointment_4_0_0', ['participant.actor', 'agent.who']],
+            ['CarePlan_4_0_0', ['careTeam.member']],
+            ['CareTeam_4_0_0', ['participant.member']],
+            ['Communication_4_0_0', ['sender', 'recipient']],
+            ['CommunicationRequest_4_0_0', ['requester', 'recipient']],
+            ['Condition_4_0_0', ['asserter']],
+            ['Consent_4_0_0', ['performer']],
+            ['DeviceRequest_4_0_0', ['requester']],
+            ['DiagnosticReport_4_0_0', ['performer']],
+            ['Encounter_4_0_0', ['participant.individual']],
+            ['EpisodeOfCare_4_0_0', ['careManager', 'team.member']],
+            ['Immunization_4_0_0', ['performer.actor']],
+            ['MedicationAdministration_4_0_0', ['performer.actor']],
+            ['MedicationRequest_4_0_0', ['requester']],
+            ['Observation_4_0_0', ['performer']],
+            ['Procedure_4_0_0', ['performer.actor']],
+            ['Provenance_4_0_0', ['agent.who']],
+            ['ServiceRequest_4_0_0', ['requester']],
+            ['SupplyRequest_4_0_0', ['requester']],
+            ['Task_4_0_0', ['requester', 'owner']]
```

Comment:
How are we determining this list of resources to update references?

Reply:
After asking Chris, he said to use ChatGPT, which I did.